design/pool-configuration.md

```diff
@@ -0,0 +1,386 @@
+# Address Pool configuration options
+
+## Summary
+
+Following @johananl's initiative in [this issue](https://github.com/metallb/metallb/issues/630)
+this enhancement tries to collect and classify various configuration degrees requested during time.
+
+## Motivation
+
+The way advanced users consume MetalLB is evolving and exposed a need for more complex configuration.
+For example, the need for isolation and multi-tenancy induced requests for being able to advertise some specific address pools from a subset of the nodes, or
+to allocate them only to a specific namespace, to mention some.
+
+### Goals
+
+* Collect and classify all the requests coming from different issues, in order to have a better understanding
+  and an organic implementation, avoiding to focus on and to implement only a specific feature, risking to cause conflicts afterwards.
+* Provide a list of the possible configuration enhancements, and to highlight dependencies and conflicts between each other (if any).
+* Highlight the cases where the solution must be BGP or L2 specific
+
+### Non-Goals
+
+* Covering only a specific use case.
+* Providing a L2 only or BGP only solution where the problem might be expressed in a more generic way.
+
+## Proposal
+
+For each ip of an address pool, we need to address three main points:
+
+* Which nodes must be advertised as destinations of that ip
+* Where / to whom the ip address must be advertised to
```

Comment:
Agree. I wasn't sure when I started writing this if we wanted to have some kind of filtering in ARP advertisement or not, ended up saying we don't in a following sectoin. Will change it.

Reply:
ARP is a broadcast-based layer 2 protocol. While it's technically possible e.g. to refuse to provide an ARP reply to a specific source MAC, doing so would be weird IMO since a security policy usually isn't enforced at that layer.